src/main/java/com/graphhopper/converter/resources/ConverterResourceGisgraphy.java

```diff
@@ -0,0 +1,199 @@
+package com.graphhopper.converter.resources;
+
+import javax.ws.rs.BadRequestException;
+import javax.ws.rs.DefaultValue;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.Response;
+
+import com.codahale.metrics.annotation.Timed;
+import com.graphhopper.converter.api.GisgraphyGeocodingResult;
+import com.graphhopper.converter.api.GisgraphySearchResult;
+import com.graphhopper.converter.api.Status;
+import com.graphhopper.converter.core.Converter;
+
+/**
+ * @author David Masclet
+ */
+@Path("/gisgraphy")
+@Produces("application/json; charset=utf-8")
+public class ConverterResourceGisgraphy extends AbstractConverterResource {
+
+	// all webservices
+	private static final String ADDRESS_PARAMETER = "address";
+	private static final String LAT_PARAMETER = "lat";
+	private static final String LONG_PARAMETER = "lng";
+	private static final String COUNTRY_PARAMETER = "country";
+	private static final String RADIUS_PARAMETER = "radius";
+	private static final String DEFAULT_FORMAT = "json";
+	private static final String FORMAT_PARAMETER = "format";
+	private static final String APIKEY_PARAMETER = "apikey";
+	// geocoding specid=fic
+	private static final String GEOCODING_LIMIT_PARAMETER = "limitnbresult";
+	// search
+	private static final String SEARCH_LIMIT_PARAMETER = "to";
+	private static final String SEARCH_QUERY_PARAMETER = "q";
+
+
+	private final String geocodingUrl;
+	private final String reverseGeocodingUrl;
+	private final String searchURL;
+	private final String apiKey;
+	private final Client jerseyClient;
+
+	public ConverterResourceGisgraphy(String geocodingUrl,
+			String reverseGeocodingUrl, String searchURL, String apiKey,
+			Client jerseyClient) {
+		this.geocodingUrl = geocodingUrl;
+		this.reverseGeocodingUrl = reverseGeocodingUrl;
+		this.searchURL = searchURL;
+		this.jerseyClient = jerseyClient;
+		this.apiKey = apiKey;
+	}
+
+	@GET
+	@Timed
+	public Response handle(@QueryParam("q") @DefaultValue("") String query,
+			@QueryParam("point") @DefaultValue("") String point,
+			@QueryParam("radius") @DefaultValue("") String radius,
+			@QueryParam("country") @DefaultValue("") String country,
+			@QueryParam("limit") @DefaultValue("5") Integer limit,
+			@QueryParam("reverse") @DefaultValue("false") boolean reverse,
+			@QueryParam("autocomplete") @DefaultValue("false") boolean autocomplete) {
```

Comment:
Should we add the locale parameter as well? The other 2 converters have a locale parameter. 

AFAIK gisgraphy allows locale input only for autocomplete searches. So adding it might not make sense? WDYT?

Reply:
no 'locale' is not managed for autocomplete (it is for the full-text web service, but not in suggest). There is a lang parameter but it has an impact on returned fields (and those fields are not returned by the suggest web service). This is not managed at all for geocoding and reverse web services. It might be an improvement but needs some code modifications on the server side that are not planned yet. We have to go without this first ;-)

Reply:
Ok no problem :), that's why I was asking.